src/main.c

```diff
@@ -1222,9 +1222,56 @@ static int on_config_max_connections(h2o_configurator_command_t *cmd, h2o_config
 
 static int on_config_num_threads(h2o_configurator_command_t *cmd, h2o_configurator_context_t *ctx, yoml_t *node)
 {
-    if (h2o_configurator_scanf(cmd, node, "%zu", &conf.num_threads) != 0)
+    size_t num_threads = 0;
+    free(conf.thread_map.entries);
+    memset(&conf.thread_map, 0, sizeof(conf.thread_map));
+    if (node->type == YOML_TYPE_SCALAR) {
+        int i;
+        if (h2o_configurator_scanf(cmd, node, "%zu", &num_threads) != 0)
+            return -1;
+        h2o_vector_reserve(NULL, &conf.thread_map, num_threads);
+        for (i = 0; i < num_threads; i++)
+            conf.thread_map.entries[conf.thread_map.size++] = -1;
+    } else if (node->type == YOML_TYPE_SEQUENCE) {
+#ifndef H2O_HAS_PTHREAD_SETAFFINITY_NP
+        fprintf(stderr, "[error] Can't handle a CPU list, this platform doesn't support `pthread_setaffinity_np`\n");
```

Comment:
I'd appreciate it if you could use `h2o_configurator_errprintf`. Unless we do so, our users might have hard time figuring out how to resolve the issue.